doc/contributing/building_ruby.md

```diff
@@ -25,17 +25,27 @@
     git clone https://github.com/ruby/ruby.git
     ```
 
-4. Generate the configuration files and build:
+4. Generate the configuration files and build. It's generally advisable to use a build directory:
 
     ```
     ./autogen.sh
-    mkdir build && cd build # its good practice to build outside of source dir
+    mkdir build && cd build # it's good practice to build outside of source dir
     mkdir ~/.rubies # we will install to .rubies/ruby-master in our home dir
     ../configure --prefix="${HOME}/.rubies/ruby-master"
     make install
     ```
 
-5. [Run tests](testing_ruby.md) to confirm your build succeeded
+5. Optional: If you are frequently building Ruby, disabling documentation will reduce the time it takes to `make install`:
+
+    ``` shell
+    ../configure --disable-install-doc
+    ```
+
+6. [Run tests](testing_ruby.md) to confirm your build succeeded
+
+### Unexplainable Build Errors
+
+If you are having unexplainable build errors, after saving all your work, try running `git clean -xfd` in the source root to remove all git ignored local files. If you are working from a source directory that's been updated several times, you may have temporary build artefacts from previous releases which can cause build failures.
```

Comment:
Typo:

```suggestion
If you are having unexplainable build errors, after saving all your work, try running `git clean -xfd` in the source root to remove all git ignored local files. If you are working from a source directory that's been updated several times, you may have temporary build artifacts from previous releases which can cause build failures.
```

Reply:
I don't mind changing this, but it's a case of [British vs American spelling](https://grammarist.com/spelling/artefact-artifact/). Do Ruby docs generally adhere to American spellings over British ones?

Reply:
TIL "artefact" is British spelling. I think the docs defaults to American English.